internal/eval/eval.go

```diff
@@ -4816,165 +4816,43 @@ func evalZRANGE(args []string, store *dstore.Store) []byte {
+	for _, op := range ops {
+		switch op.Kind {
 		case GET:
-			if len(args) <= i+2 {
-				return diceerrors.NewErrWithMessage(diceerrors.SyntaxErr)
-			}
-			eType, eVal, offset, err := parseEncodingAndOffset(args[i+1 : i+3])
-			if err != nil {
-				return diceerrors.NewErrWithFormattedMessage(err.Error())
-			}
-			ops = append(ops, BitFieldOp{
-				Kind:   GET,
-				EType:  eType.(string),
-				EVal:   eVal.(int64),
-				Offset: offset.(int64),
-				Value:  int64(-1),
-			})
-			i += 3
+			res := value.getBits(int(op.Offset), int(op.EVal), op.EType == SIGNED)
+			result = append(result, res)
 		case SET:
-			if len(args) <= i+3 {
-				return diceerrors.NewErrWithMessage(diceerrors.SyntaxErr)
-			}
-			eType, eVal, offset, err := parseEncodingAndOffset(args[i+1 : i+3])
-			if err != nil {
-				return diceerrors.NewErrWithFormattedMessage(err.Error())
-			}
-			value, err1 := strconv.ParseInt(args[i+3], 10, 64)
-			if err1 != nil {
-				return diceerrors.NewErrWithMessage(diceerrors.IntOrOutOfRangeErr)
-			}
-			ops = append(ops, BitFieldOp{
-				Kind:   SET,
-				EType:  eType.(string),
-				EVal:   eVal.(int64),
-				Offset: offset.(int64),
-				Value:  value,
-			})
-			i += 4
+			prevValue := value.getBits(int(op.Offset), int(op.EVal), op.EType == SIGNED)
+			value.setBits(int(op.Offset), int(op.EVal), op.Value)
+			result = append(result, prevValue)
 		case INCRBY:
-			if len(args) <= i+3 {
-				return diceerrors.NewErrWithMessage(diceerrors.SyntaxErr)
-			}
-			eType, eVal, offset, err := parseEncodingAndOffset(args[i+1 : i+3])
+			res, err := value.incrByBits(int(op.Offset), int(op.EVal), op.Value, overflowType, op.EType == SIGNED)
 			if err != nil {
-				return diceerrors.NewErrWithFormattedMessage(err.Error())
-			}
-			value, err1 := strconv.ParseInt(args[i+3], 10, 64)
-			if err1 != nil {
-				return diceerrors.NewErrWithMessage(diceerrors.IntOrOutOfRangeErr)
+				result = append(result, nil)
+			} else {
+				result = append(result, res)
 			}
-			ops = append(ops, BitFieldOp{
-				Kind:   INCRBY,
-				EType:  eType.(string),
-				EVal:   eVal.(int64),
-				Offset: offset.(int64),
-				Value:  value,
-			})
-			i += 4
 		case OVERFLOW:
-			if len(args) <= i+1 {
-				return diceerrors.NewErrWithMessage(diceerrors.SyntaxErr)
-			}
-			switch strings.ToUpper(args[i+1]) {
-			case WRAP, FAIL, SAT:
-				overflowType = strings.ToUpper(args[i+1])
-			default:
-				return diceerrors.NewErrWithFormattedMessage(diceerrors.OverflowTypeErr)
-			}
-			ops = append(ops, BitFieldOp{
-				Kind:   OVERFLOW,
-				EType:  overflowType,
-				EVal:   int64(-1),
-				Offset: int64(-1),
-				Value:  int64(-1),
-			})
-			i += 2
-		default:
-			return diceerrors.NewErrWithMessage(diceerrors.SyntaxErr)
+			overflowType = op.EType
 		}
 	}
+	return result
+}
+
+// Generic method for both BITFIELD and BITFIELD_RO.
+// isReadOnly method is true for BITFIELD_RO command.
+func evalBITFIELDGeneric(args []string, store *dstore.Store, isReadOnly bool) []byte {
```

Comment:
Let's reserve the eval prefix for top level eval commands. We can rename this as follows.
```suggestion
func bitfieldEvalHelper(args []string, store *dstore.Store, isReadOnly bool) []byte {
```